### DISCUSSION TITLE
Using SeleniumBase in UC mode

### DISCUSSION THREAD

**Discussion by benfrankie** [seleniumbase/SeleniumBase#2322] [Q&A] [answered]
Hi, 

1. Does SeleniumBase automatically set the user agent in UC mode? If so, is it randomized for each session, or does it remain the same across different sessions?
Is there any risk of detection if the same user agent is used across multiple sessions when running multiple accounts?
I noticed the same user agent appearing on all browsers. 

2. For use cases involving multiple accounts, is it advisable to manually set unique user agents for each account to avoid detection? If yes, could you provide guidance on how to implement this maybe with fake_useragent while using UC mode?

3. Does SeleniumBase create a fresh browser instance without utilizing browser profiles or cache for each session? If so, could this lack of consistent browser data across sessions increase the risk of account flagging or detection?
Would using consistent browser profiles with SeleniumBase mitigate detection risks when automating multiple account sessions?
If so, how can I set up browser profiles in UC mode?

4. Where' can I find a comprehensive manual or detailed instructions for using SeleniumBase in UC mode without the class-based approach?
I have scripts that use the standard WebDriver, undetected ChromeDriver, and CloudScraper, but all of them fail to bypass Cloudflare.  I've noticed that SB successfully bypasses it. The issue I have with SB is that some standard codes, such as 'if __name__ == "__main__":', setting up a user agent or opening/closing a new tab, don't seem to work. Please help! 

5. I've encountered an issue where elements on an infinite scroll page stop loading after scrolling down for like 10 min, particularly when I switch between tabs. This issue occurs when using an undetected driver. Can SeleniumBase help resolve this, and if so, how?

Thanks!

**Comment by mdmintz** [comment, marked as answer]
Many of your questions may already be answered in the UC Mode video tutorial:

<b>https://www.youtube.com/watch?v=5dMFI3e85ig</b>

<a href="https://www.youtube.com/watch?v=5dMFI3e85ig"><img src="https://github.com/seleniumbase/SeleniumBase/assets/6788579/4c9a112a-25c6-458f-83c6-c4b910b8b636" title="undetected automation" width="330" /></a>

SeleniumBase fixes the user agent in headless mode by swapping `HeadlessChrome` with just `Chrome`. Otherwise, Chrome uses the natural browser's user agent, which is the best way to go because changing that could lead to an inconsistency with the browser's natural fingerprint (which can get you detected).

If you want to set a `user_data_dir`, you can. This lets you keep a Chrome profile between tests. But be sure you never mix a UC Mode `user_data_dir` with one that is used by a regular Chrome instance, because that will get you detected.

Here are a few of the UC Mode methods: (More are covered in the video)

```python
driver.uc_open_with_reconnect(url, reconnect_time=None)
driver.uc_click(selector)
```

Here are some examples of UC Mode scripts:

* https://github.com/seleniumbase/SeleniumBase/blob/master/examples/verify_undetected.py
* https://github.com/seleniumbase/SeleniumBase/blob/master/examples/uc_cdp_events.py
* https://github.com/seleniumbase/SeleniumBase/blob/master/examples/raw_uc_mode.py

Here's are some newer UC Mode examples where clicking is required:

* https://github.com/seleniumbase/SeleniumBase/blob/master/examples/raw_turnstile.py
* https://github.com/seleniumbase/SeleniumBase/blob/master/examples/raw_form_turnstile.py

(There are [23 SeleniumBase Syntax Formats](https://github.com/seleniumbase/SeleniumBase/blob/master/help_docs/syntax_formats.md) for structuring scripts.)

Most methods scroll automatically on most pages. There are also some specialized scroll methods:

```python
self.scroll_to(selector)
self.slow_scroll_to(selector)
self.scroll_into_view(selector)
self.scroll_to_top()
self.scroll_to_bottom()
```
